src/main/java/io/vertx/core/net/ContextualBindException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2011-2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.net;
+
+import java.net.BindException;
+import java.net.SocketAddress;
+
+/**
+ * An exception that is meant to stand in for {@link BindException} and provide information
+ * about the target which caused the bind exception.
+ */
+public class ContextualBindException extends RuntimeException {
```

Comment:
which not extend BindException ?

Reply:
BinException is not a RuntimeException, but in this case, we could extend BindException.

Reply:
Because `BindException` does not provide a constructor that takes a cause

Reply:
you can still all `initCause(Throwable)` @geoand

Reply:
not using an exception extending BindException would be a breaking change

Reply:
Huh, I never knew about `initCause`...

Thanks! Changed